packages/knip/src/plugins/astro/index.ts

```diff
@@ -12,14 +12,26 @@ const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependenc
 
 export const config = ['astro.config.{js,cjs,mjs,ts,mts}'];
 
-const entry = ['src/content/config.ts', 'src/content.config.ts'];
+const resolveEntryPaths: ResolveEntryPaths = async (_, options) => {
+  const { configFilePath } = options;
+  const astroConfig = await load(configFilePath);
+  const srcDir = astroConfig?.srcDir ?? 'src';
 
-const production = [
-  'src/pages/**/*.{astro,mdx,js,ts}',
-  'src/content/**/*.mdx',
-  'src/middleware.{js,ts}',
-  'src/actions/index.{js,ts}',
-];
+  return [`${srcDir}/content/config.ts`, `${srcDir}/content.config.ts`].map(path => toEntry(path));
+};
+
+const resolveConfig: ResolveConfig = async (_, options) => {
+  const { configFilePath } = options;
+  const astroConfig = await load(configFilePath);
```

Comment:
The `resolveConfig` function is executed with `_` being the `astroConfig` so there would be no need to also load that separately.

However, the  `resolveEntryPaths` is meant to handle (production) entry paths. Same issue there :)

Please see https://knip.dev/guides/writing-a-plugin#6-resolveconfig and https://knip.dev/guides/writing-a-plugin#7-resolveentrypaths

Reply:
FYI, I have refactored plugins so that we only need to implement `resolveConfig` (`resolveEntryPaths` is removed). This is also updated in the docs: https://knip.dev/guides/writing-a-plugin#example-2-config. However, in https://github.com/webpro-nl/knip/pull/1067#discussion_r2072604186 I've suggested to use `resolveFromAST` so eventually it should not matter for this PR.